src/components/app/index.tsx

```diff
@@ -93,6 +98,19 @@ export default class App extends Component<Props, State> {
     const fileInput = event.target as HTMLInputElement;
     const sourceFile = fileInput.files && fileInput.files[0];
     if (!sourceFile) return;
+    await this.updateFile(sourceFile);
+  }
+
+  @bind
+  async onFileDrop(event: CustomEvent) {
+    const sourceFile = event.detail as File;
+    this.setState(initialState());
+
+    if (!sourceFile) return;
```

Comment:
Should this check appear before resetting state?

Reply:
good catch.

Reply:
done.